lib/internal/modules/esm/get_format.js

```diff
@@ -73,6 +74,9 @@ function extname(url) {
  * @returns {string}
  */
 function getFileProtocolModuleFormat(url, context, ignoreErrors) {
+  if (isESModule)
+    return 'module';
```

Comment:
So this would load _any_ file as an ES module? IIUC that means `import './package.json'` would try to load `package.json` as an ES module and not a JSON module.

Reply:
Yep, there should be some sort of `&& isMain` check